File: src/1299-Replace-Elements-with-Greatest-Element-on-Right-Side/1299.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int max_right = -1;
        for (int i = arr.length - 1; i >= 0; i--) {
            max_right = Math.max(arr[i], arr[i] = max_right);
        }
        return arr;
    }
}

File: src/1299-Replace-Elements-with-Greatest-Element-on-Right-Side/1299.go
func replaceElements(arr []int) []int {
    max_right := -1
    for i := len(arr) - 1; i >= 0; i-- {
        max_right, arr[i] = max(arr[i], max_right), max_right
    }
    return arr
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}